TicTacToeGame.cpp

```diff
@@ -0,0 +1,97 @@
+#include <iostream>
+#include <iomanip>
+
+void initializeBoard(int a[3][3]) {
+    for(int i = 0; i < 3; i++) {
+        for(int j = 0; j < 3; j++) {
+            a[i][j] = 0;
+        }
+    }
+}
+
+bool isOccupied(int row, int col, int a[3][3]) {
+    return a[row][col] != 0;
+}
+
+bool isOutOfBounds(int row, int col) {
+    return (row < 0 || row > 2 || col < 0 || col > 2);
+}
+
+bool isNotAvailable(int row, int col, int a[3][3]) {
+    return (isOutOfBounds(row, col) || isOccupied(row, col, a));
+}
+
+void enterASymbol(int a[3][3], int player) {
+    int row, col;
+    do{
+        if(player == 1) std::cout << "It's X turn! Please enter your coordinates: " << std::endl;
+        else std::cout << "It's O turn! Please enter your coordinates: " << std::endl;
+        std::cin >> row >> col;
+        row--;
+        col--;
+        if(!isNotAvailable(row, col, a)) {
+            a[row][col] = player;
+            break;
+        } else std::cout << "Sorry, but that space isn't available / doesn't exist! Try again! " << std::endl;
+    } while(isNotAvailable(row, col, a));
```

Comment:
Try to optimize this check - You are checking a position two times on every iteration, you can do it with only one check.
While(true) is an option